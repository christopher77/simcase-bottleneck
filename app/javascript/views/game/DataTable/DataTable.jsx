import React from "react";
import styles from "./DataTable.module.css";
import Station from "../../../components/Station/Station";
import classNames from "classnames";

function DataTable(props) {
  const { stations, demand, feedback, flowRate } = props;
  const dataInfo = Object.values(stations);

  return (
    <div className={styles.dataTableView}>
      {!feedback && (
        <h4 className={styles.message}>
          Identify the bottleneck and selectively add capacity. Consider
          throughput and click on the square to assign resources.
        </h4>
      )}
      <div className={styles.tableContainer}>
        <h4 className={styles.title}>Stations</h4>
        <div
          className={classNames(styles.list, {
            [styles.listFeedback]: feedback
          })}
        >
          {dataInfo.map((info, index) => {
            return (
              <Station
                key={index}
                info={info}
                flowRate={flowRate}
                feedback={feedback}
                isCorrect={
                  feedback === false ||
                  stations[`station${index + 1}`].machinesAssigned ===
                    stations[`station${index + 1}`].solution
                }
              />
            );
          })}
        </div>
        <div className={styles.demandContainer}>
          <h3 className={styles.tons}>
            {demand}
            <span> {" lbs"} </span>
          </h3>
          <h4 className={styles.demand}>demand</h4>
        </div>
      </div>
    </div>
  );
}

export default DataTable;
