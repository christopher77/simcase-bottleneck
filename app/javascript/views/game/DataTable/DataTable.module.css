.dataTableView {
  height: inherit;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  display: grid;
  grid-template: 64px 1fr / 1fr;
  height: inherit;
  width: auto;
}

.message {
  text-align: center;
  margin: 0 21px 10px;
  color: #4f4f4f;
  font-weight: 400;
  font-size: 14px;
}

.tableContainer {
  align-self: normal;
}

.title {
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  margin-left: 21px;
}

.list {
  overflow: auto;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 55.8vh;
}
.listFeedback {
  height: 61.8vh;
}

.demandContainer {
  line-height: 30px;
  display: flex;
  text-align: center;
  flex-direction: column;
  align-items: center;
  justify-self: center;
  margin-bottom: 15px;
  margin-top: -3px;
}

.tons {
  font-weight: bold;
  font-size: 30px;
  line-height: 37px;
  width: 150px;
}
.tons > span {
  font-weight: bold;
  font-size: 30px;
  line-height: 37px;
  white-space: pre;
}

.demand {
  text-transform: capitalize;
  width: 150px;
  font-weight: 500;
  font-size: 14px;
  line-height: 10px;
}

/* .list::-webkit-scrollbar {
  display: none;
} */

@media (min-width: 680px) {
  .demandContainer {
    padding-top: 5%;
  }
}
