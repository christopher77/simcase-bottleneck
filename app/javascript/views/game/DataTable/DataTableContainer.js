import { connect } from "react-redux";
import DataTable from "./DataTable";
import flowRate from "../../../components/utils/Bottleneck/FlowRate";
import { getRoundCapacities } from "../../../redux/selectors/index";

const mapStateToProps = (state, ownProps) => {
  const roundNumber = `round${ownProps.round}`;
  const demand = state.rounds.data.Demand;
  const currentFlowRate = flowRate(demand, getRoundCapacities(state));

  return {
    stations: state.rounds[roundNumber].stations,
    demand: state.rounds[roundNumber].demand,
    feedback: state.feedback.feedback,
    flowRate: currentFlowRate
  };
};

export default connect(
  mapStateToProps,
  null
)(DataTable);
