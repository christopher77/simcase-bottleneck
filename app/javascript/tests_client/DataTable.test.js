import React from "react";
import { render } from "@testing-library/react";
import DataTable from "../views/game/DataTable/DataTable";

test("DataTable view", () => {
  const { asFragment } = render(<DataTable stations={{}} />);
  expect(asFragment).toMatchSnapshot();
});
