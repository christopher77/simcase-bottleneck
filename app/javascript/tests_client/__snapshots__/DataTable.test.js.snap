// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable view 1`] = `[Function]`;
